samples/unit-tests/series-flags/general/demo.js

```diff
@@ -201,4 +201,36 @@ QUnit.test('Scrolling inverted chart with a flag series.', function (assert) {
         false,
         'The flag series\' DOM elements should not contain NaN attributes values (#14063).'
     );
-});
\ No newline at end of file
+});
+
+QUnit.test('Distributing the flag, #16041.)', function (assert) {
+    const chart = Highcharts.chart('container', {
+        chart: {
+            width: 800
+        },
+        series: [{
+            data: [
+                [1, 10],
+                [6, 60]
+            ],
+            type: 'line'
+        }, {
+            data: [{
+                x: 5,
+                title: "Very long long text very long long text"
+            }],
+            type: "flags"
+        }]
+    });
+
+    assert.ok(
+        chart.series[1].points[0].graphic.x > 0,
+        'The flag graphic should have a positive x position.'
+    );
+    chart.setSize(400, undefined);
```

Comment:
The second argument `undefined` is the default one, not needed here, but it can stay of course.